tests/test_routing.py

```diff
@@ -1159,3 +1159,36 @@ async def startup() -> None:
             ...  # pragma: nocover
 
         router.on_event("startup")(startup)
+
+
+def host_endpoint(request):
+    name = request.path_params["name"]
+    return JSONResponse({"name": name})
+
+
+host_based_router = Router(
+    routes=[
+        Host(
+            "example.org",
+            app=Router([Route("/foo/{name}", host_endpoint, name="foo")]),
```

Comment:
Is the name necessary?

Reply:
I wanted to make sure the correct endpoint is handling the request so in addition to checking the status_code check the response.json() too.

Reply:
If you delete the name, doesn't the test passes?

Reply:
I mean the argument name, not the path parameter.

Reply:
Makes sense, I removed it.